### DISCUSSION TITLE
Is any built function to convert AssertionPipeline to smt2 format?

### DISCUSSION THREAD

**Discussion by fraktlz** [cvc5/cvc5#11016] [Q&A] [answered]
Is any built function to convert AssertionPipeline to smt2 format? I want to dump the AssertionPipeline in https://github.com/cvc5/cvc5/blob/main/src/preprocessing/passes/non_clausal_simp.cpp. Thank you.

**Comment by ajreynol** [comment, marked as answer]
Yes, this class is made for this purpose: https://github.com/cvc5/cvc5/blob/main/src/smt/print_benchmark.h.  All you need is to extract the `std::vector<Node>` corresponding to the input formulas and defined functions and that class should print it as a valid *.smt2.
